### ISSUE TITLE
awlsimhw_linuxcnc: Error on loading program to CPU 

### ISSUE THREAD

**Issue by uncle-yura** [mbuesch/awlsim#19]
When gui reloads the program, the instance variable hal is cleared. The hal.ready () function is also executed twice, which causes an error.

```
Traceback (most recent call last):
  File "/home/lcnc/awlsim/examples/linuxcnc-demo/../..//awlsim-linuxcnc-hal", line 438, in <module>
    sys.exit(main())
  File "/home/lcnc/awlsim/examples/linuxcnc-demo/../..//awlsim-linuxcnc-hal", line 394, in main
    server.run()
  File "/home/lcnc/awlsim/awlsim/coreserver/server.py", line 1683, in run
    self.__handleCommunicationBlocking()
  File "/home/lcnc/awlsim/awlsim/coreserver/server.py", line 1506, in __handleCommunicationBlocking
    self.__handleCommunication()
  File "/home/lcnc/awlsim/awlsim/coreserver/server.py", line 1466, in __handleCommunication
    self.__handleSocketComm(rlist)
  File "/home/lcnc/awlsim/awlsim/coreserver/server.py", line 1439, in __handleSocketComm
    self.__handleClientComm(self.__sock2client[sock])
  File "/home/lcnc/awlsim/awlsim/coreserver/server.py", line 1423, in __handleClientComm
    handler(self, client, msg)
  File "/home/lcnc/awlsim/awlsim/coreserver/server.py", line 1049, in __rx_RUNSTATE
    self.setRunState(self.STATE_RUN)
  File "/home/lcnc/awlsim/awlsim/coreserver/server.py", line 642, in setRunState
    raise e
AttributeError: 'unicode' object has no attribute 'ready'

```

I came up with adding a global variable for hal instance.

In file "awlsim/coreserver/server.py" added
`304: self.linuxcncHALcomponent = None`
and
```
961: if "linuxcnc" in hwmodName:
hwmodDesc.setParameterValue('hal',self.linuxcncHALcomponent)
```

In file "awlsim_linuxcnc_hal"
`359: server.linuxcncHALcomponent = hal`

In file "awlsimhw_linuxcnc/main.py"
```
207: if not self.LinuxCNC_HAL.component_is_ready("awlsim"):
self.hal.ready()
```

**Comment by mbuesch** [comment]
Thanks for your bug report.

**Comment by mbuesch** [comment]
Fixed in Awlsim 0.72.0.
